MadeByMunstersLLC/fuel
# Feature add tags and category management 12 22 16

**Changes:**

* Fixes spec/dummy app for testing
* Adds dotenv for dev testing aws and sample.env
* Adds single migration to handle all table creations and column changes with up and down methods
* Adds CRUD interface for Tag Management
* Adds CRUD interface for Category Management
* Updates Post Form to include new Category and Tag Selects
* Updates icons for side-nav
* Adds Tags and Category to json returned from posts index and show

**Testing:**
* Pull down the repo and `bundle install`
* cd into `spec/dummy`
* create `.env` from `sample.env`
* add a test AWS account `not really used for testing`
* run `db:create && db:migrate`
* run `rails s`
* Test and make sure that you can create/update/delete Tags and Categories
* Test and make sure you can attach Tags and Category to a Post
* Test and make sure that deleting Tag and Category removes associations join table
* Test and make sure JSON returned has both `tags` and `category`

**Notes:**
For updating schlep we will need to run the rake task below to copy the migration over to Schlep.
```rake fuel:install:migrations```

**Trello Links:**
https://trello.com/c/wnuGi0Gh/952-add-category-management-to-fuel-gem
https://trello.com/c/t8Dd7j0h/953-add-tag-management-to-fuel-gem
